src/components/modes/teacher/AvatarDialog.js

```diff
@@ -307,9 +373,44 @@ class AvatarDialog extends Component {
     dispatchCloseAvatarDialog();
   };
 
+  handleOnClickUserChecked = (value) => {
+    const { avatar } = this.state;
+    const { userList } = avatar;
+    const currentIndex = userList.indexOf(value);
+    // make a copy of the array
+    const newUserList = [...userList];
+
+    // not found
+    if (currentIndex === -1) {
+      newUserList.push(value);
+    } else {
+      newUserList.splice(currentIndex, 1);
+    }
```

Comment:
Maybe explicitly add a comment when the user is removed and when the user is added, for clarity.